File: fx/src/io/github/gravetii/controller/GameResultController.java
package io.github.gravetii.controller;

import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.TableColumn;
import javafx.scene.control.TableRow;
import javafx.scene.control.TableView;
import javafx.scene.control.cell.PropertyValueFactory;
import javafx.scene.input.MouseEvent;

import static io.github.gravetii.controller.GameResultDisplayer.TableResult;

public class GameResultController implements FxController {

  private final GameGridController ref;
  private GameResultDisplayer displayer;

  @FXML private TableView<TableResult> tblDisplay;
  @FXML private TableColumn<TableResult, String> wordTblCol;
  @FXML private TableColumn<TableResult, Integer> pointsTblCol;

  public GameResultController(GameGridController ref) {
    this.ref = ref;
  }

  @FXML
  public void initialize() {
    this.wordTblCol.setCellValueFactory(new PropertyValueFactory<>("word"));
    this.pointsTblCol.setCellValueFactory(new PropertyValueFactory<>("score"));
    this.tblDisplay.setRowFactory(
        callback -> {
          TableRow<TableResult> row = new TableRow<>();
          row.setOnMouseClicked(
              event -> {
                if (event.getClickCount() == 2 && !row.isEmpty()) {
                  TableResult result = row.getItem();
                  String word = result.getWord();
                  if (result.isByUser()) {
                    this.ref.revisitUserWord(word);
                  } else {
                    this.ref.revisitGameWord(word);
                  }
                }
              });

          return row;
        });

    this.displayer = new GameResultDisplayer(this.tblDisplay);
  }

  @FXML
  public void rotate(MouseEvent event) {
    this.ref.rotate();
  }

  @FXML
  public void onGoBtnClick(ActionEvent event) {
    this.ref
        .validateWordOnBtnClick()
        .ifPresent(
            result -> {
              this.displayer.showUserWord(result);
            });
  }
}

File: src/main/kotlin/io/github/gravetii/controller/GameResultController.kt
package io.github.gravetii.controller

import io.github.gravetii.model.TableResult
import javafx.fxml.FXML
import javafx.scene.control.TableColumn
import javafx.scene.control.TableRow
import javafx.scene.control.TableView
import javafx.scene.control.cell.PropertyValueFactory

class GameResultController(private val ref: GameGridController) : FxController {

    private lateinit var displayer: GameResultDisplayer

    @FXML
    private lateinit var tblDisplay: TableView<TableResult>

    @FXML
    private lateinit var idTblCol: TableColumn<TableResult, Int>

    @FXML
    private lateinit var wordTblCol: TableColumn<TableResult, String>

    @FXML
    private lateinit var pointsTblCol: TableColumn<TableResult, Int>

    @FXML
    fun initialize() {
        idTblCol.cellValueFactory = PropertyValueFactory("id")
        wordTblCol.cellValueFactory = PropertyValueFactory("word")
        pointsTblCol.cellValueFactory = PropertyValueFactory("score")
        tblDisplay.setRowFactory {
            val row = TableRow<TableResult>()
            row.setOnMouseClicked {
                if (!row.isEmpty) {
                    val result = row.item
                    result.word?.let {
                        if (result.byUser) ref.revisitUserWord(it)
                        else ref.revisitGameWord(it)
                    }
                }
            }
            row
        }

        displayer = GameResultDisplayer(tblDisplay)
    }

    @FXML
    fun rotate() = ref.rotate()

    @FXML
    fun onGoBtnClick() {
        ref.validateWordOnBtnClick()?.let {
            displayer.showUserWord(it)
            displayer.setMarker()
        }
    }

}